src/custom-elements/src/components/phone-input/phone-input.tsx

```diff
@@ -170,6 +170,14 @@ export class ChiPhoneInput {
     this.chiChange.emit(this.value);
   };
 
+  __validate = evt => {
+    const charCode = evt.which ? evt.which : evt.keyCode;
+
+    if (charCode > 31 && (charCode < 48 || charCode > 57)) evt.preventDefault();
+
+    return true;
+  };
+
```

Comment:
I did some research and found that the `which` and `keyCode` properties are deprecated, so I made another approach using the property `code` .

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/which
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code

If you see, for each event we attach a handler not a function. With that handler we can add more things to do in the future whereas with the function it is limited.

```suggestion
  _keyPressHandler = (event: KeyboardEvent): void => {
    this._checkNoCharactersInInput(event);
  }

  _checkNoCharactersInInput(event : KeyboardEvent) {
    const inputKey = event.code;

    if (!inputKey.includes('Digit')) {
      event.preventDefault();
    }
  };

```

Reply:
@josecarlosgonzalezv this looks good to me. Thanks